### DISCUSSION TITLE
How to use Keda to scale an HTTP API service comparable to Knative?

### DISCUSSION THREAD

**Discussion by Spenhouet** [kedacore/keda#5500] [Q&A / Need Help] [answered]
How would you use Keda to scale an API service? 

Currently we are using Knative and it automatically buffers HTTP requests and auto scales depending on the number of requests, but Knative has some black-box automagic behavior which makes us want to switch away. 
We are currently exploring Keda in the area of event driven where Keda scales based on the RabbitMQ queue. Keda seems straight forward and transparent in what it does. Looks like a great choice. But we fail to see how we could utilize Keda for an API service. We probably would need a HTTP request buffer component again. A form of proxy queue for HTTP requests.

**Comment by zroubalik** [comment, marked as answer]
Hi, you can check https://github.com/kedacore/http-add-on

**Reply by Spenhouet** [reply]
We will definitely check that out. A bit sad that it's declared as not-production-ready. We would definitely need it for production.

**Reply by zroubalik** [reply]
We are open to contributions to make it more stable :)

**Reply by Spenhouet** [reply]
Not sure if we are going to be able to contribute via PRs but should we look into it, we will definitely contribute by testing and providing feedback.

**Reply by zroubalik** [reply]
@JorTurFer FYI

**Reply by JorTurFer** [reply]
Hey @Spenhouet ,
Thanks for thinking in KEDA xD
Currently we are working on improving the add-on until GA, but sadly it won't happen soon. It doesn't mean that you can't use it on prod, but probably you will need to work a bit more around it and take more things into account. Recently we have opened this issue to track the thinks that we want to address before going to GA: https://github.com/kedacore/http-add-on/issues/911

For example, we are testing it internally (in my company) for some prod workloads. The reason for being in beta is that we think that it doesn't have enough quality (yet) according to our standards or we think that there are missing features, but it doesn't mean that you can't use it or that it'll fail randomly and/or catastrophically. 

If you are still interested on using the add-on, I suggest testing it in non prod envs to ensure that it works for you and your scenario and giving us feedback (please, open the issues in the add-on repo instead of here). 
If you test the add-on and it works for you in your use case, being or not being in GA doesn't matter as you can expect the same support level xD I mean, in both cases, the support that we offer is community best effort (there are enterprise vendors, but this is the OSS project)

**Reply by Spenhouet** [reply]
@JorTurFer Thanks for the encouragement to try it out. We are currently doing so in a Proof of Concept.
Right now my colleague is struggling with getting everything setup and he mentioned that the documentation doesn't seem to be up-to-date. He will open issues on the http-add-on repo for these. I'm still optimistic that it's just things we missed.

**Reply by JorTurFer** [reply]
Hey!
Sure, any issue that you see in docs please open a PR there and we will try to solve it. The problem there is that docs aren't versioned on that repo, and maybe your folk is checking the next version docs because main branch doc is for main branch code but the latest version released is 0.7.0 (and there are some changes related with metrics and aggregation windows).
For HTTPScaledObject reference you should use `https://github.com/kedacore/http-add-on/blob/main/docs/ref/v0.7.0/http_scaled_object.md` until v0.8.0 is released (I hope at the end of the week). Once v0.8.0 is available, I'd suggest migrating to it directly before adding any workload more because until v0.8.0 there was a bug with helm chart that forces to remove add-on deployments to upgrade the chart
https://github.com/kedacore/http-add-on/issues/973

**Reply by Manikiran88888** [reply]
Hey Good Morning, 
I have used the same code which is mentioned above but could not scale the pods after the providing the load for the http requests 

This is the link to mu issue 
https://github.com/kedacore/http-add-on/issues/1005 
There you have asked for the logs and I have shared the logs please check it and suggest me some solution
